From 6b339ec04028d09f5039487dee048ad1501b7aba Mon Sep 17 00:00:00 2001
From: b07117 <b07117@freescale.com>
Date: Mon, 20 Feb 2012 22:42:47 +0800
Subject: [PATCH 238/498] ENGR00174836 correct Android YUV format mess usage

HAL_PIXEL_FORMAT_YCbCr_420_P use 3 YUV planes, same to I420
HAL_PIXEL_FORMAT_YCbCr_420_I use 1 plane with YUV interleaved

Signed-off-by: Li Xianzhong <b07117@freescale.com>
---
 mx6/libgralloc/framebuffer.cpp | 2 +-
 mx6/libgralloc/gralloc.cpp     | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/mx6/libgralloc/framebuffer.cpp b/mx6/libgralloc/framebuffer.cpp
index d9d4910..1431675 100755
--- a/mx6/libgralloc/framebuffer.cpp
+++ b/mx6/libgralloc/framebuffer.cpp
@@ -533,7 +533,7 @@ static int fb_post(struct framebuffer_device_t* dev, buffer_handle_t buffer)
 
 static int fb_compositionComplete(struct framebuffer_device_t* dev)
 {
-    glFinish();
+   // glFinish();
     return 0;
 }
 
diff --git a/mx6/libgralloc/gralloc.cpp b/mx6/libgralloc/gralloc.cpp
index 3b96f24..0b3e3a9 100755
--- a/mx6/libgralloc/gralloc.cpp
+++ b/mx6/libgralloc/gralloc.cpp
@@ -368,6 +368,9 @@ static int gralloc_alloc(alloc_device_t* dev,
         return -EINVAL;
 
     if (m->gpu_device && !(usage & GRALLOC_USAGE_HWC_OVERLAY)) {
+       if(format == HAL_PIXEL_FORMAT_YCbCr_420_I)format = HAL_PIXEL_FORMAT_YCbCr_420_P;
+       else if(format == HAL_PIXEL_FORMAT_YCbCr_420_P)format = HAL_PIXEL_FORMAT_YCbCr_420_I;
+
        return m->gpu_device->alloc(dev, w, h, format, usage, pHandle, pStride);
     }
 
-- 
1.8.0

